executor/joiner.go

```diff
@@ -222,12 +222,7 @@ func (j *baseJoiner) filterAndCheckOuterRowStatus(input, output *chunk.Chunk, in
 	}
 
 	// Batch copies selected rows to output chunk.
-	innerColOffset, outerColOffset := 0, innerColsLen
-	if !j.outerIsRight {
-		innerColOffset, outerColOffset = input.NumCols()-innerColsLen, 0
-	}
-
-	_, err = chunk.CopySelectedJoinRows(input, innerColOffset, outerColOffset, j.selected, output)
+	_, err = chunk.CopySelectedJoinRowsDirect(input, j.selected, output)
```

Comment:
Does each inner row is the same here?
Could we implement and use  `CopySelectedJoinRowsWithSameInnerRows` here?

Reply:
Yes, but in practice, the improvement is rare in a query's execution, even though the benchmark on small size data is better. This is like the benefits from vectorized expression.